Find a functionally equivalent code:android.hardware.SystemSensorManager.registerListenerImpl:COMMENT
Method Modifier: protected   hidden      
Comment:/**
 * @hide
 */

@@ -4,10 +4,10 @@
         return false;
     }
     // Trigger Sensors should use the requestTriggerSensor call.
-    if (Sensor.getReportingMode(sensor) == Sensor.REPORTING_MODE_ONE_SHOT) {
+    if (sensor.getReportingMode() == Sensor.REPORTING_MODE_ONE_SHOT) {
         Log.e(TAG, "Trigger Sensors should use the requestTriggerSensor.");
         return false;
     }

